### PR TITLE
[PFX-403] Gasket: Update upgrade guide (v7)

### PR DESCRIPTION
**Pull request by jpina1-godaddy**
## Summary

Add upgrade guide docs for the v7/ version of Gasket.

## Changelog

Added upgrade docs for v7

### CODE REVIEW

File: docs/upgrade-to-7.md [hunk 1]
```diff
@@ -0,0 +1,132 @@
...
+We had some lifecycles that did not conform to the context object pattern we have in place for many other lifecycles.
+
+The reason for utilizing this context object is to enable the execution of these lifecycle methods under two distinct scenarios:
+
+1. During build time, when there is no request object available.
+2. During run time, when the request object becomes available.
+
+Affected lifecycles:
+
+- `appRequestConfig`
```
Comment (agerard-godaddy) [inline docs/upgrade-to-7.md:78]: Another big change was renaming config plugin to response-data plugin ([#680](https://github.com/godaddy/gasket/pull/680)). Although yes, this lifecycle context has been normalized, it is under a new name now: `responseData` ([doc change](https://github.com/godaddy/gasket/pull/680/files#diff-cffc29cc7d0e75f6e9ea8cde81073208211edbea7c8de39c341574d318541304R243)).

You could probably leave it out of this section, but add a new section for the config -> response-data changes.

File: docs/upgrade-to-7.md [hunk 2]
```diff
@@ -0,0 +1,132 @@
...
+- `initReduxState`
+- `nextPreHandling` `.d.ts` type file
+
+[(#669)]
+
+If your app or plugins hooks these lifecycles you may need to adjust them.
+
+```diff
+- async appRequestConfig(gasket, config, req, res) {
++ async appRequestConfig(gasket, config, { req, res }) {
```
Comment (agerard-godaddy) [inline docs/upgrade-to-7.md:88]: Use a different lifecycle hook for the example, per the previous comment.

File: docs/upgrade-to-7.md [hunk 3]
```diff
@@ -0,0 +1,132 @@
...
+```
+
+## @gasket/plugin-elastic-apm
+
+- Add setup script to create hook.
+- Remove sensitive cookie filter and config.
```
Comment (agerard-godaddy) [inline docs/upgrade-to-7.md:94]: We were able to retain the sensitive cookie filter, right @bbetts-godaddy?

File: docs/upgrade-to-7.md [hunk 4]
```diff
@@ -0,0 +1,132 @@
...
+
+[(#672)]
+
+## Bring Your Own Logger
```
Comment (agerard-godaddy) [inline docs/upgrade-to-7.md:98]: This section needs work with examples, which I'm presuming we'll have a separate ticket for? - Just double checking considering it has more details that the previous sections which also need some more detail work.

Comment (jpina1-godaddy) [reply docs/upgrade-to-7.md:98]: Yeah I was going to create a ticket for this. I don't have all the context on what was changed what steps are need to take in order to upgrade.

**Review by mmason2-godaddy — APPROVED** [review]

**Review by agerard-godaddy — APPROVED** [review]

**Pull request merged**